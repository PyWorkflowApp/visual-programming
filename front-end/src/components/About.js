import React from 'react';
import { Modal } from 'react-bootstrap';

export default class About extends React.Component {

    constructor(props) {
      super(props);
      this.state = {
          show: props.show,
          info: {}};
    }

    componentDidMount() {

    }

    fetchInfo = async () => {
        const resp = await fetch("/info");
        const data = await resp.json();
        this.setState({info: data});
    }

    handleClose = () => {
      this.setState({show: false});
    }

    handleShow = (e) => {
        e.preventDefault();
        this.fetchInfo()
        this.setState({show: true});
    }

    render() {
      return (
          <>
              <div><span className="btn btn-link" onClick={this.handleShow}>About</span></div>

              <Modal show={this.state.show} onHide={this.handleClose}>
                  <Modal.Header closeButton>
                      <Modal.Title><b>About Visual Programming</b></Modal.Title>
                  </Modal.Header>
                  <Modal.Body>
                      {JSON.stringify(this.state.info)}
                  </Modal.Body>
              </Modal>
          </>
      );
    }
}
