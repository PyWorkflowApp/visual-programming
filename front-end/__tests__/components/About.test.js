import React from 'react'
import { render } from '@testing-library/react'
import ReactDOM from 'react-dom';
import About from '../../src/components/About';

global.console = {error: jest.fn()}

global.fetch = jest.fn(() => Promise.resolve({
  data: [],
  json: jest.fn(() => [])
}));

describe('Validates About', () => {
  it('Does not display About info', () => {
    const div = React.createElement('div');
    const app = render(<About show={false} />, div);
    expect(app).toMatchSnapshot();
  });

  it('Displays About info', () => {
    const div = React.createElement('div');
    const app = render(<About show={true} />, div);
    expect(app).toMatchSnapshot();
  });

  it('Validates closing', () => {
    const props = {
      show: true
    };

    const event = {
      preventDefault: jest.fn(() => [])
    };

    const about = new About(props);
    about.componentDidMount();
    about.handleShow(event);
    about.handleClose();

    expect(event.preventDefault.mock.calls.length).toBe(1);
  });
});
