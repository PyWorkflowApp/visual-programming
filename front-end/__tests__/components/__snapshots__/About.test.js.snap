// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validates About Displays About info 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    style="padding-right: 0px;"
  >
    <div
      aria-hidden="true"
    >
      <div>
        <span
          class="btn btn-link"
        >
          About
        </span>
      </div>
    </div>
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <div
              class="modal-title h4"
            >
              <b>
                About Visual Programming
              </b>
            </div>
            <button
              class="close"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
              <span
                class="sr-only"
              >
                Close
              </span>
            </button>
          </div>
          <div
            class="modal-body"
          >
            {}
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div>
      <span
        class="btn btn-link"
      >
        About
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Validates About Does not display About info 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <span
          class="btn btn-link"
        >
          About
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <span
        class="btn btn-link"
      >
        About
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
